src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -73,9 +73,13 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
                 if (!inputSchema.TryFindColumn(inputColumnName, out var originalColumn))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", inputColumnName);
 
-                if ((originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector ||
-                   originalColumn.Kind == SchemaShape.Column.VectorKind.Vector) && Transformer.ValueColumnType is VectorType)
-                    throw Host.ExceptNotSupp("Column '{0}' cannot be mapped to values when the column and the map values are both vector type.", inputColumnName);
+                if (originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector ||
+                   originalColumn.Kind == SchemaShape.Column.VectorKind.Vector)
+                {
+                    if (Transformer.ValueColumnType is VectorType)
+                        throw Host.ExceptNotSupp("Column '{0}' cannot be mapped to values when the column and the map values are both vector type.", inputColumnName);
```

Comment:
>when the column and the map values are both vector type. [](start = 90, length = 56)

i am not sure i understand this . could you give an example where this condition would be hit ?

Reply:
This is in a situation where column={"a", "b", "c"} and map={"a": [1,2,3], "b":[4,5], "c":[6]}
the resulting column will be output_col={[1,2,3], [4,5], [6]}


---
In reply to: [253629485](https://github.com/dotnet/machinelearning/pull/2386#discussion_r253629485) [](ancestors = 253629485)

Reply:
thanks for the example.

so we shouldn't support  a scenario  where Transformer.ValueColumnType is VectorType
i.e   map={"a": [1,2,3], "b":[4,5], "c":[6]}

but then i do not understand  why we need the  condition in line #76-#77 ?  

if input is not vector :   
e.g.  
column="a"  => output_col = {[1,2, 3]}
column="b" =>  output_col = {[4,5]}

maybe i am missing something ?

---
In reply to: [253656971](https://github.com/dotnet/machinelearning/pull/2386#discussion_r253656971) [](ancestors = 253656971,253629485)

Reply:
We do support the scenario where Transformer.ValueColumnType is VectorType. However, we don't support scenario where both input and Transformer.ValueColumnType is VectorType are of vector type. 

---
In reply to: [253665022](https://github.com/dotnet/machinelearning/pull/2386#discussion_r253665022) [](ancestors = 253665022,253656971,253629485)